File: apps/api/app/schemas/billing.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


PlanName = Literal["free", "pro", "family"]


class CheckoutSessionCreate(BaseModel):
    plan: Literal["pro", "family"]


class CheckoutSessionResponse(BaseModel):
    checkout_url: str


class BillingPortalResponse(BaseModel):
    portal_url: str


class BillingStatusResponse(BaseModel):
    plan_name: PlanName
    subscription_status: str
    is_paid: bool
    stripe_customer_id: str | None
    stripe_subscription_id: str | None
    current_period_end: datetime | None
    cancel_at_period_end: bool

File: apps/web/src/types/billing.ts
export type BillingPlan =
  | "free"
  | "pro"
  | "family";

export type PaidBillingPlan =
  | "pro"
  | "family";

export type BillingStatus = {
  plan_name: BillingPlan;
  subscription_status: string;
  is_paid: boolean;
  stripe_customer_id: string | null;
  stripe_subscription_id: string | null;
  current_period_end: string | null;
  cancel_at_period_end: boolean;
};

export type CheckoutSessionResponse = {
  checkout_url: string;
};

export type BillingPortalResponse = {
  portal_url: string;
};